File: Property_management_system/src/com/property/management/Tenant.java
package com.property.management;

import java.util.ArrayList;
import java.util.List;

public class Tenant {
    private String tenantID;
    private String name;
    private String contactInfo;
    private String leaseID;
    public static List<Tenant> tenants = new ArrayList<>();

    public Tenant(String tenantID, String name, String contactInfo, String leaseID) {
        this.tenantID = tenantID;
        this.name = name;
        this.contactInfo = contactInfo;
        this.leaseID = leaseID;
    }

    // Getters and Setters
    public String getTenantID() {
        return tenantID;
    }

    public void setTenantID(String tenantID) {
        this.tenantID = tenantID;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getContactInfo() {
        return contactInfo;
    }

    public void setContactInfo(String contactInfo) {
        this.contactInfo = contactInfo;
    }

    public String getLeaseID() {
        return leaseID;
    }

    public void setLeaseID(String leaseID) {
        this.leaseID = leaseID;
    }

    // Methods
    public static void addTenant(Tenant tenant) {
        tenants.add(tenant);
        System.out.println("Tenant added: " + tenant.name);
    }

    public static void editTenant(String tenantID, String newName, String newContactInfo, String newLeaseID) {
        for (Tenant tenant : tenants) {
            if (tenant.tenantID.equals(tenantID)) {
                tenant.setName(newName);
                tenant.setContactInfo(newContactInfo);
                tenant.setLeaseID(newLeaseID);
                System.out.println("Tenant updated: " + tenant.name);
                return;
            }
        }
        System.out.println("Tenant not found!");
    }

    public static void deleteTenant(String tenantID) {
        tenants.removeIf(tenant -> tenant.tenantID.equals(tenantID));
        System.out.println("Tenant deleted: " + tenantID);
    }
}

File: src/tenant.py
class Tenant:
    tenants = []

    def __init__(self, tenant_id, name, contact_info, lease_id):
        self.tenant_id = tenant_id
        self.name = name
        self.contact_info = contact_info
        self.lease_id = lease_id

    @classmethod
    def add_tenant(cls, tenant):
        cls.tenants.append(tenant)
        print(f"Tenant {tenant.name} added.")

    @classmethod
    def edit_tenant(cls, tenant_id, new_name, new_contact_info, new_lease_id):
        for tenant in cls.tenants:
            if tenant.tenant_id == tenant_id:
                tenant.name = new_name
                tenant.contact_info = new_contact_info
                tenant.lease_id = new_lease_id
                print(f"Tenant {tenant_id} updated.")
                return
        print("Tenant not found!")

    @classmethod
    def delete_tenant(cls, tenant_id):
        cls.tenants = [
            tenant for tenant in cls.tenants if tenant.tenant_id != tenant_id]
        print(f"Tenant {tenant_id} deleted.")
